### PR TITLE
[fix]: [PLG-2319]: block CI trial NG licenses

### PR DESCRIPTION
**Pull request by kapil-harness**
## Describe your changes
block CI trial NG licenses

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47464)

### CODE REVIEW

**Comment by kapil-harness** [comment]
trigger uts

**Review by devinder-harness — APPROVED** [review]

File: 930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java [hunk 1]
```diff
@@ -261,9 +263,16 @@ public ResponseDTO<ModuleLicenseDTO> startCommunityLicense(
       @io.swagger.v3.oas.annotations.parameters.RequestBody(required = true,
           description = "This is the details of the Trial License. ModuleType and edition are mandatory") @NotNull
       @Valid @Body StartTrialDTO startTrialRequestDTO) {
+    checkCITrialLicense(startTrialRequestDTO);
     return ResponseDTO.newResponse(licenseService.extendTrialLicense(accountIdentifier, startTrialRequestDTO));
   }
 
+  private void checkCITrialLicense(StartTrialDTO startTrialRequestDTO) {
+    if (startTrialRequestDTO.getModuleType().equals(ModuleType.CI)) {
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java:271]: lets do the `ModuleType.CI == startTrialRequestDTO.getModuleType()` instead to keep it null-safe check ?

**Review by neelam-harness — APPROVED** [review]

**Comment by kapil-harness** [comment]
trigger uts

**Review by kapil-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47464)* status: 3 file(s) to review. 1. unresolved discussions. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java` line 271 at r2](https://reviewable.io/reviews/harness/harness-core/47464#gh-1187057045:-NUu50nhB88oH39FfEVQ:b3taky5) ([raw file](https://github.com/harness/harness-core/blob/fabe55d8fa897a779a1299d9ec1526ab64ba6600/930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java#L271)):*
<details><summary><i>Previously, neelam-harness wrote…</i></summary><blockquote>

lets do the `ModuleType.CI == startTrialRequestDTO.getModuleType()` instead to keep it null-safe check ? 
</blockquote></details>

I have changed it to ModuleType.CI.equals(startTrialRequestDTO.getModuleType()) and it is NULL safe.

File: 930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java [hunk 2]
```diff
@@ -261,9 +263,16 @@ public ResponseDTO<ModuleLicenseDTO> startCommunityLicense(
       @io.swagger.v3.oas.annotations.parameters.RequestBody(required = true,
           description = "This is the details of the Trial License. ModuleType and edition are mandatory") @NotNull
       @Valid @Body StartTrialDTO startTrialRequestDTO) {
+    checkCITrialLicense(startTrialRequestDTO);
     return ResponseDTO.newResponse(licenseService.extendTrialLicense(accountIdentifier, startTrialRequestDTO));
   }
 
+  private void checkCITrialLicense(StartTrialDTO startTrialRequestDTO) {
+    if (ModuleType.CI.equals(startTrialRequestDTO.getModuleType())) {
```
Comment (hemanthmantri) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java:271]: Nit: Keep this method generic. Have a list of modules for which this shouldn't be supported and check if getModuleType is one of those. For now, CI is the only element in that list.

**Review by hemanthmantri — COMMENTED** [review]
Non blocking comment.

**Pull request merged**